compiler/src/iree/compiler/Codegen/Common/GPU/GPUNestedLayoutDistributionPatterns.cpp

```diff
@@ -466,6 +475,61 @@ struct DistributeTransferWrite final
           writeOp, "warp or thread tiles have overlapping strides");
     }
 
+    // If the distribution results in threads writing to the same address, guard
+    // with an scf.if to ensure only one thread writes per duplication group.
+
+    // Delinearize the thread id into
+    //   ('workgroup_overlap', *subgroup_tile, 'lane_overlap', *thread_tile)
+    SmallVector<int64_t> basis(2 * rank + 1);
+    ArrayRef<int64_t> subgroupTile = vectorLayout.getSubgroupTile();
+    ArrayRef<int64_t> threadTile = vectorLayout.getThreadTile();
+    int64_t threadTileSize = std::accumulate(
+        threadTile.begin(), threadTile.end(), 1, std::multiplies<int64_t>());
+    int64_t laneOverlap = std::max<int64_t>(1, subgroupSize / threadTileSize);
```

Comment:
nit: it's generally discouraged to request a specific overload of standard library functions like this because you don't know what the other template arguments might be
```suggestion
    int64_t laneOverlap = std::max(int64_t(1), subgroupSize / threadTileSize);
```